src/script/descriptor.cpp

```diff
@@ -212,6 +212,12 @@ struct PubkeyProvider
 
     /** Derive a private key, if private data is available in arg. */
     virtual bool GetPrivKey(int pos, const SigningProvider& arg, CKey& key) const = 0;
+
+    /** Return all (extended) public keys for this PubkeyProvider
+     * param[out] pubkeys Any public keys
+     * param[out] ext_pubs Any extended public keys
+     */
+    virtual void GetRootPubKeys(std::set<CPubKey>& pubkeys, std::set<CExtPubKey>& ext_pubs) const = 0;
```

Comment:
In 8fcd9cfb4c1dac680e9c6702b7af9517a578210d

I think `PubkeyProvider` can hold only 1 (extended) public key.  So should we have a different signature for the method?
Do I miss something?

Reply:
Good point, changed to two functions `GetRootPubKey` and `GetRootExtPubKey`, each returning an optional `CPubKey` or `CExtPubKey`, respectively.